cloud/google/clusteractuator.go

```diff
@@ -67,3 +120,66 @@ func getOrNewComputeServiceForCluster(params ClusterActuatorParams) (GCEClientCo
 	}
 	return computeService, nil
 }
+
+func (gce *GCEClusterClient) createFirewallRuleIfNotExists(cluster *clusterv1.Cluster, firewallRule *compute.Firewall) error {
+	ruleExists, ok := cluster.ObjectMeta.Annotations[firewallRuleAnnotationPrefix + firewallRule.Name]
+	if ok && ruleExists == "true" {
+		// rule was already created, so just return
+		return nil
+	}
+	clusterConfig, err := gce.clusterproviderconfig(cluster.Spec.ProviderConfig)
+	if err != nil {
+		return fmt.Errorf("Error parsing cluster provider config: %v", err)
```

Comment:
nit: avoid full sentences/capitalization at the start for error messages.
https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences

Reply:
done.